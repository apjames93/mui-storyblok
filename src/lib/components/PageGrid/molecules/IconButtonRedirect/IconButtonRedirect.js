import React from 'react';
import PropTypes from 'prop-types';
import { validComponentsRequired } from 'lib/utils/customProps';
import appRedirect from 'lib/utils/appRedirect';
import IconButton from '../IconButton/IconButton';

/**
 * IconButtonRedirect is used in storyblok redirect to react routes
 */

export const IconButtonRedirect = ({
  href,
  iconButton,
  external,
  callback,
}) => {
  const externalRedirect = () => {
    window.location.replace(href);
  };

  const internalRedirect = () => {
    appRedirect(href);
  };

  const onClick = () => {
    if (callback) callback();
    if (external) {
      return externalRedirect();
    }
    return internalRedirect(href);
  };

  const btn = iconButton[0];

  return (
    <IconButton
      {...btn}
      onClick={onClick}
    />
  );
};

export default IconButtonRedirect;

IconButtonRedirect.propTypes = {
  /** redirect route */
  href: PropTypes.string.isRequired,
  /** IconButton Allowed maximum: 1 */
  iconButton(props, propName, componentName) {
    return validComponentsRequired(props, propName, componentName, ['IconButton'], 1);
  },
  // callback to run before redirect
  callback: PropTypes.func,
  /**
   * if false will redirect to a page in the app and should use '/page-whatever'
   *
   *  if true will redirect to another site 'https://www.google.com/'
   * */
  external: PropTypes.bool,
};

IconButtonRedirect.defaultProps = {
  iconButton: [],
  callback: undefined,
  external: false,
};
