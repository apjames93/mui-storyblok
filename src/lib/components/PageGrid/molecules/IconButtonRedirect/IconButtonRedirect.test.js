import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import { IconButtonRedirect } from './IconButtonRedirect';

function setup(legacyRedirect = true) {
  const props = {
    redirectRoute: '/page-welcome',
    legacyRedirect,
    iconButton: [{
      component: 'IconButton',
      onClick: jest.fn(),
      icon: [{
        component: 'Icon',
        iconName: 'star',
      }],
    }],

    callback: jest.fn(),

  };
  const comp = mount(<IconButtonRedirect {...props} />);

  return { comp, props };
}

describe('<IconButtonRedirect />', () => {
  it('renders IconButtonRedirect', () => {
    const { comp } = setup();
    expect(comp).toBeDefined();
  });

  describe('clicks', () => {
    it('handleClick and calls history push ', async () => {
      const { comp } = setup(false);
      window.location.assign = jest.fn();
      comp.find('WithStyles(ForwardRef(IconButton))').at(0).simulate('click');
      expect(window.location.assign).toBeCalled();
    });
  });

  test('snapshot', () => {
    const { props } = setup();

    const tree = renderer.create(<IconButtonRedirect {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
