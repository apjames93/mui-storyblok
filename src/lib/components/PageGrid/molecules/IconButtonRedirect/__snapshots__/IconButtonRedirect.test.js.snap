// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IconButtonRedirect /> snapshot 1`] = `
<button
  className="MuiButtonBase-root MuiIconButton-root makeStyles-root-7"
  data-testid="muiIconButton"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiIconButton-label"
  >
    <span
      aria-hidden={true}
      className="material-icons MuiIcon-root makeStyles-root-8 undefined MuiIcon-colorSecondary"
    >
      star
    </span>
  </span>
</button>
`;
